refactor(ast): add `NULL` value for serializing null types

camchenry opened the pull request:
from https://github.com/oxc-project/oxc/pull/8980#discussion_r1948222912, added a special value that just represents a unit type, which should always be serialized to `null`

camchenry commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8987?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8987** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8987?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8987?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8980** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8980?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8987).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hi-ogawa reviewed (commented):
Nice! There are two more `value = ()` in `crates/oxc_ast/src/ast/literal.rs`. Can we change that too?

overlookmotel commented:
> Nice! There are two more `value = ()` in `crates/oxc_ast/src/ast/literal.rs`. Can we change that too?

In the interests of merging this swiftly, I've pushed an extra commit to do that.

camchenry committed:
refactor(ast): add `NULL` value for serializing null types (#8987)

from https://github.com/oxc-project/oxc/pull/8980#discussion_r1948222912, added a special value that just represents a unit type, which should always be serialized to `null`
